pkg/scheduler/schedule_one_test.go

```diff
@@ -448,6 +487,136 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 	}
 }
 
+func TestSchedulerGuaranteeNonNilNodeInSchedulingCycle(t *testing.T) {
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var (
+		initialNodeNumber        = 2000
+		initialPodNumber         = 500
+		waitSchedulingPodNumber  = 2000
+		deleteNodeNumberPreRound = 20
+		createPodNumberPreRound  = 25
+
+		fakeSchedulerName = "fake-scheduler"
+		fakeNamespace     = "fake-namespace"
+
+		initialNodes []runtime.Object
+		initialPods  []runtime.Object
+	)
+
+	for i := 0; i < initialNodeNumber; i++ {
+		nodeName := fmt.Sprintf("node%d", i)
+		initialNodes = append(initialNodes, st.MakeNode().Name(nodeName).UID(nodeName).Obj())
+	}
+	// randomly scatter initial pods onto nodes
```

Comment:
Stylish: could you make the first letter uppercase and ends with period? which is sort of (although not 100%) conventional in scheduler's comments. (applies elsewhere)

```suggestion
	// Randomly scatter initial pods onto nodes.
```

Reply:
of course